### PR TITLE
fix: Fire the received event on linux

### LINKED ISSUE #20: Deeplink is not working on Linux/Ubuntu
Deeplink is not working on Linux/Ubuntu.

Installed and integrated plugin into our electron project.
open on chrome with :
<a href="myapp://param">Open app</a>

getting below error
linux_deeplink_issue

### PR DESCRIPTION
**Pull request by Joelkang**
Because the `open-url` and `open-file` events are [macOS only](https://www.electronjs.org/docs/latest/api/app#event-open-file-macos),
we still need to rely on the second-instance event handler for
linux (in addition to win32).

Fixes #20

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -81,18 +80,18 @@ class Deeplink extends EventEmitter {
 
         if (os.platform() === 'darwin') {
             app.setAsDefaultProtocolClient(protocol);
+
+            app.on('will-finish-launching', () => {
```
Comment (Joelkang) [inline src/index.ts:84]: moving this up, so that it's wildly clear that this is darwin only

File: src/index.ts [hunk 2]
```diff
@@ -154,7 +155,7 @@ class Deeplink extends EventEmitter {
         }
     };
 
-    private openEvent = (event: any, url: string, eventName: string) => {
+    private darwinOpenEvent = (event: any, url: string, eventName: string) => {
```
Comment (Joelkang) [inline src/index.ts:158]: ditto for renaming this in favour of clarity that this is darwin only

**Comment by Joelkang** [comment]
@glawson do you think you have time to take a look at this PR? Not rushing, just trying to plan for whether I should localise these change for now

**Review by glawson — APPROVED** [review]
LGTM!!

**Pull request merged**